flocker/acceptance/test_api.py

```diff
@@ -432,3 +432,44 @@ def not_exists():
             return deleted
         created.addCallback(delete_dataset)
         return created
+
+
+    def test_dataset_grow(self):
+        """
+        The size of a dataset can be increased.
+        """
+        # Create a dataset with REALISTIC_BLOCKDEVICE_SIZE
+        # Reconfigure that dataset to be REALISTIC_BLOCKDEVICE_SIZE * 2
+        # Check for 200 response code
+        # http://doc-dev.clusterhq.com/advanced/api.html#post--v1-configuration-datasets-%28dataset_id%29
+        # Wait for the state of that dataset to show the updated size.
+        1/0
+
+    def test_dataset_shrink(self):
+        """
+        The size of a dataset can be decreased.
+        """
+        # Create a dataset with REALISTIC_BLOCKDEVICE_SIZE * 2
+        # Reconfigure that dataset to be REALISTIC_BLOCKDEVICE_SIZE
+        # Check for 200 response code
+        # Wait for the state of that dataset to show the updated size.
+        1/0
+
+    def test_dataset_shrink_not_possible(self):
+        """
+        If the dataset has too much data to allow shrinking to the requested size...then what??
```

Comment:
The convergence attempts will fail ... indefinitely ... or until some data is removed.  It will be hard to write a test for this because we don't have a convergence failure reporting mechanism yet (and one isn't planned for 1.0).  I suggest skipping testing this at the API level (we should perhaps test it at the convergence agent level though).